### ISSUE TITLE
issues with CompletionItemResolve

### ISSUE THREAD

**Issue by bwkam** [haskell/haskell-language-server#4451]
## Your environment 

Which OS do you use? 
**NixOS**

Which version of GHC do you use and how did you install it?
**9.6.6, with nix**

How is your project built (alternative: link to the project)?

Which LSP client (editor/plugin) do you use?
**emacs+lsp-mode**
Which version of HLS do you use and how did you install it?
**2.9.0.0, using nix**
Have you configured HLS in any way (especially: a `hie.yaml` file)?
**No**

## What's wrong?

some error about SMethod_CompletionItemResolve keeps popping up as I type anything in the buffer
the error is not an emacs problem too, same behavior occurs on vscode (I just tested)
(see below)
### Debug information

```
Error processing message (error "No plugins are available to handle this SMethod_CompletionItemResolve request.
 Plugins installed for this method, but not available to handle this request are:
ghcide-completions does not handle resolve requests for (unable to determine resolve owner)).").
```

**Comment by fendor** [comment]
Thank you for the bug report!

I can not reproduce on arch linux with HLS 2.9.0.1 installed via ghcup.
However, I can reproduce the issue with a nixpkgs installed HLS binary.

**Comment by dmitrykvasnikov** [comment]
I have the same issue in doomemacs, emacs+lsp-mode , vanill arch, ghc 9.10.1

**Comment by meritamen** [comment]
Same here, vanilla emacs+lsp-haskell, macOS Ventura 13.7.1, ghc 9.4.8, hls 2.9.0.1

**Comment by bradrn** [comment]
I’m seeing the same issue in vanilla Emacs with lsp-haskell, Arch Linux (well, EndeavourOS), GHC 9.4.8, HLS 2.1.0.1

**Comment by fendor** [comment]
Reading into the code, I don't think there is anything wrong, sometimes additional information can be requested, and sometimes the server doesn't have any additional information. E.g., documentation or type signatures for non-local identifiers. Is the current behaviour annoying, e.g., does Emacs or vim spam you with errors?

**Comment by bwkam** [comment]
yeah in emacs, its extremely annoying. a buffer keeps popping up as you type

**Comment by fendor** [comment]
But this must be a new change in emacs, right? I think we had this behaviour for a while and just now we started to get these reports?

Seemingly, HLS is behaving correctly. I think this issue is resolvable by extending `CompletionResolveData` with a `NothingToResolve` and inserting as the default for all completion items.

**Comment by bradrn** [comment]
> Is the current behaviour annoying, e.g., does Emacs or vim spam you with errors?

Correct. I get a message every time the autocomplete event fires, i.e. every time I press a key (or nearly so).

**Comment by fendor** [comment]
I created #4463 which I would sneak into #4448 to get it out ASAP.

Does someone want to double check this works?

**Comment by dmitrykvasnikov** [comment]
> Does someone want to double check this works?
If you tell me what to do - i'll try :-)

**Comment by fendor** [comment]
If you use GHCup, the following works:

```
ghcup compile hls -g d51c68dcf7a418bde4d02f62345710f298168328 --ghc 9.6.6
```

Or whatever GHC version you use in your project.

You can also use clone the project, and run `cabal install`, then point emacs to your freshly installed HLS version.

**Comment by dmitrykvasnikov** [comment]
> If you use GHCup, the following works:
> 
> ```
> ghcup compile hls -g d51c68dcf7a418bde4d02f62345710f298168328 --ghc 9.6.6

I did this for ghc 9.10.1
This message now doesn't appear when auto-completion list consist of haskell only keywords / modules etc, but if it includes one of my function in it - this message still continue to appear. Screenshots below

![1733836143](https://github.com/user-attachments/assets/80a12759-8ed4-474e-9d4f-bfed6676d558)
![1733836130](https://github.com/user-attachments/assets/61636870-68b0-4bef-98d4-11f60783a4b5)

**Comment by fendor** [comment]
Thanks, another test to add to the testsuite!

**Comment by fendor** [comment]
@dmitrykvasnikov Where was the local function defined you tried to complete? In the same module, different module, etc...? I seem unable to reproduce the issue.

**Comment by dmitrykvasnikov** [comment]
@fendor 
In this same module (it's written above on screenshot).  Just tried with function defined in different module - works fine, no message.
Just now I did as example:
**func :: Int -> Int -> Int
func = (+)**
So if after that i put in my code:
**g = fu ..** - in this moment autocomplete gives me suggestions (with function defined above) and error message appears.

**Comment by fendor** [comment]
Can't reproduce, I am getting the correct message
```
Trace - 17:31:55] Received response 'completionItem/resolve - (20)' in 2ms.
Result: {
    "data": {
        "itemFile": "file:///home/hugin/Documents/haskell/hls-completion-resolve/A.hs",
        "itemName": [
            "A",
            "main-398bd560290e6bafe7a4d6959f05d45a6937707a",
            "v",
            "func"
        ],
        "itemNeedsType": false,
        "tag": "CompletionResolveData"
    },
    "detail": ":: Int -> Int -> Int",
    "documentation": {
        "kind": "markdown",
        "value": "*Defined at line 3, column 1 in this module*\n"
    },
    "insertText": "func",
    "insertTextFormat": 2,
    "kind": 3,
    "label": "func",
    "sortText": "00"
}
```

Maybe the compiled binary wasn't used? I am double checking my instructions.

**Comment by fendor** [comment]
I am a little bit confused :sweat_smile: After running `ghcup compile ...`, does `ls -alh ~/.ghcup/bin/haskell-language-server-9.10.1~2.9.0.1` hint that the binary was freshly created?
Does `~/.ghcup/bin/haskell-language-server-9.10.1` point to `~/.ghcup/bin/haskell-language-server-9.10.1~2.9.0.1`?

**Comment by bwkam** [comment]
I'm compiling this myself to see if I can reproduce (taking a while)

**Comment by bwkam** [comment]
ok, so if I'm doing this correctly. the error still pops up, but less frequently. not so helpful footage of when it triggers attached

using ghc 9.6.6

checked the binary is freshly created
```
>ls -alh ~/.local/bin/haskell-language-server
Permissions Size User  Date Modified Name
lrwxrwxrwx     - bwkam 10 Dec 20:08   /home/bwkam/.local/bin/haskell-language-server -> ../state/cabal/store/ghc-9.6.5/haskell-language-server-2.9.0.1-e-haskell-language-server-bda016429039356b5e7ae15d0b54c3a626539f8f90dccf348e7458730e7bef3d/bin/haskell-language-server
```

![Peek 2024-12-10 20-30](https://github.com/user-attachments/assets/20cee35e-9f6b-44bf-ba2c-2c05753c7964)

**Comment by dmitrykvasnikov** [comment]
> I am a little bit confused 😅 After running `ghcup compile ...`, does `` hint that the binary was freshly created? Does `~/.ghcup/bin/haskell-language-server-9.10.1` point to `~/.ghcup/bin/haskell-language-server-9.10.1~2.9.0.1`?

I've just checked - yes to both questions. haskell-language-server-wrapper updated as well (if it's relevant) .

**Comment by fendor** [comment]
Very puzzling... the Completions shown by @bwkam look completely wrong, too. We should never suggest LANGUAGE pragmas in this context as a possible completion?!

**Comment by bwkam** [comment]
yeah, I was not sure why it was suggesting that

**Comment by briancarbone** [comment]
Seeing the same error; completions fail with an error, but code actions succeed and insert the import as expected.
